geps/gep-1897.md

```diff
@@ -125,7 +132,251 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+assume the connection is TLS, and verify that the TargetRef’s certificate can
+be validated by the provided trusted CA certificates before the connection is made.  On the question of how to signal
+that there was a failure in the certificate validation, this is left up to the implementation to return a response error
+that is appropriate, such as one of the HTTP error codes: 400 (Bad Request), 401 (Unauthorized), 403 (Forbidden), or
+other signal that makes the failure sufficiently clear to the requester without revealing too much about the transaction,
+based on established security requirements.
+
+All policy resources must include TargetRef with the fields specified 
+[here](https://github.com/kubernetes-sigs/gateway-api/blob/a33a934af9ec6997b34fd9b00d2ecd13d143e48b/apis/v1alpha2/policy_types.go#L24-L41).
+In an upcoming extension to TargetRef, policy resources _may_ also choose to include `SectionName` and/or `Port` in the
+TargetRef following the same mechanics as `ParentRef`.
+
+Not covered here, but possible to add would be additional configuration options mentioned in [SIG-NET Gateway API: TLS
+to the K8s.Service/Backend](https://docs.google.com/document/d/1RTYh2brg_vLX9o3pTcrWxtZSsf8Y5NQvIG52lpFcZlo).  All of
+these are currently implementation-dependent, with the following recommended defaults:
+
+- Server Name Indication: allow passing of the server name through server name indication, in the TLS transaction, to
+assist with selection of certificates when several hosts share the same IP address. (default to enabled)
+- Subject Alternative Name certificates: allow authentication with a single certificate that can serve multiple domain
+names that are intended to represent the same endpoint.  For example, these five SAN hosts: example.com, 
+www.example.com, example.net, IPv4 x.x.x.x, and IPv6 x.x.x.x.x.x.x.x all resolve to the same endpoint.
+(default to enabled)
+- Version: specifies the minimum TLS version that the connection may use (default TLSv1.2)
+- Ciphers: specifies enabled ciphers to be used in TLS exchanges. (default to align with default TLS version chosen for
+Version, e.g. ciphers mentioned for TLS version 1.1, 1.2, or 1.3 as described in 
+[RFC4346 TLS 1.1](https://datatracker.ietf.org/doc/html/rfc4346#appendix-A.5),
+[RFC 5246 TLS 1.2](https://datatracker.ietf.org/doc/html/rfc5246#appendix-A.5), or
+[RFC 8446 TLS 1.3](https://datatracker.ietf.org/doc/html/rfc8446#appendix-B.4).
+- Visibility: the visibility level could be `all` or `none`, and represents that if the connection failed, 
+the implmentation would drop the connection silently if the visibility level were `none`, or report an error if the
+visibility level were `all`.  (default to `all`)
+
+Thus, the following additions would be made to the Gateway API:
+
+```go
+// TLSConnectionPolicy provides a way to publish TLS configuration
+// that enables a gateway client to connect to a backend pod.
+type TLSConnectionPolicy struct {
```

Comment:
https://github.com/kubernetes-sigs/gateway-api/pull/2159#discussion_r1259991693 - seems like everyone was voting to not use `Policy` in the name. Posting here for visibility.

@robscott @howardjohn

Reply:
Good callout - I had the same point. Is the `...Policy` here okay?

Reply:
Both of these are metaresources that use the Policy Attachment pattern,, so I think that having `...Policy` should actually be a requirement. I've commented on the other issue to the same effect.

Reply:
If anyone can come up with a better name, I welcome feedback, but I think that having something that clearly signals "this object is a metaresource that uses Policy Attachment" in the name is a must-have.

Reply:
Since virtually all configs in K8S are also Policies - how does it help if all have this suffix ? Policy is not specific to Gateways, there are authz policies, etc. 

I don't mind if it is decided that all Gateway API resources have a suffix and is used in all, for discoverability. But maybe not a broad word as 'policy', and it must be clearly defined in the spec what is the word and that all resources must use it.

Reply:
In the interest of time, I kindly request that you open a new issue to discuss naming policies.